src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
> That doesn't matter, because rsGrabReg() might spill an in-use register.

Yes, that's right. That would seem to imply that the register should also be unspilled by the GS cookie code when it is no longer needed.

> Why do you prefer your solution instead of mine, which isolates the change to the GS cookie code, and doesn't require a global variable?

As mentioned, your solution does not isolate the change to the GS cookie code: that code still needs to be aware of which registers are in use. Any solution needs to mark the appropriate registers used and unused.

> Note that your change currently doesn't work for varargs, since the code to reload the varargs reg arguments beyond the fixed reg args doesn't update the `fixedArgsMask`

I am aware: AFAIK that should be a non-issue for .NET Core, as we do not support varargs.